blockchain/state_manager/src/utils.rs

```diff
@@ -100,3 +109,270 @@ where
         .load_sector_infos(&*state_manager.get_block_store(), &not_proving)
         .map_err(|err| Error::Other(format!("failed to get proving set :{:}", err)))
 }
+
+pub fn get_miner_sector_set<DB>(
+    state_manager: &StateManager<DB>,
+    tipset: &Tipset,
+    address: &Address,
+    filter: &mut Option<&mut BitField>,
+    filter_out: bool,
+) -> Result<Vec<ChainSectorInfo>, Error>
+where
+    DB: BlockStore,
+{
+    let miner_actor_state: miner::State = state_manager
+        .load_actor_state(&address, &tipset.parent_state())
+        .map_err(|err| {
+            Error::State(format!(
+                "(get sset) failed to load miner actor state: {:}",
+                err
+            ))
+        })?;
+    load_sectors_from_set(
+        state_manager.get_block_store_ref(),
+        &miner_actor_state.sectors,
+        filter,
+        filter_out,
+    )
+}
+
+fn load_sectors_from_set<DB>(
+    block_store: &DB,
+    ssc: &Cid,
+    filter: &mut Option<&mut BitField>,
+    _filter_out: bool,
+) -> Result<Vec<ChainSectorInfo>, Error>
+where
+    DB: BlockStore,
+{
+    let amt = Amt::load(ssc, block_store).map_err(|err| Error::Other(err.to_string()))?;
+
+    let mut sset: Vec<ChainSectorInfo> = Vec::new();
+    let for_each = |i: u64, sector_chain: &miner::SectorOnChainInfo| -> Result<(), String> {
+        if let Some(ref mut s) = filter {
+            let i = i
+                .try_into()
+                .map_err(|_| "Could not convert from index to usize")?;
+            if s.get(i) {
+                return Ok(());
+            }
+        }
+        sset.push(ChainSectorInfo {
+            info: sector_chain.info.to_owned(),
+            id: i,
+        });
+        Ok(())
+    };
+    amt.for_each(for_each)
+        .map_err(|err| Error::Other(format!("Error Processing ForEach {:}", err)))?;
+
+    Ok(sset)
+}
+
+pub fn miner_sector_info<DB>(
+    state_manager: &StateManager<DB>,
+    address: &Address,
+    sector_number: &SectorNumber,
+    tipset: &Tipset,
+) -> Result<Option<SectorOnChainInfo>, Error>
+where
+    DB: BlockStore,
+{
+    let miner_actor_state: miner::State = state_manager
+        .load_actor_state(&address, &tipset.parent_state())
+        .map_err(|err| {
+            Error::State(format!(
+                "(get sset) failed to load miner actor state: {:}",
```

Comment:
That is the accurate error message 

https://github.com/filecoin-project/lotus/blob/9b1d2f4c137433c1fe1f5ed44ba2e70105a409f0/chain/stmgr/utils.go#L149

Reply:
So these error messages don't need to match Lotus for any reason, and I assume they just made this inconsistency by accident. All I'm saying is that it would be easier to debug an issue if one of those errors is hit.

Fine to leave as is though, this is just opinionated